xwiki-commons-core/xwiki-commons-xml/src/main/java/org/xwiki/xml/XMLUtils.java

```diff
@@ -314,20 +401,39 @@ public static String minimalEscape(Object content)
      * Escapes the XML special characters in a <code>String</code> using numerical XML entities, so that the resulting
      * string can safely be used as an XML text node. Specifically, escapes &lt;, &gt;, and &amp;.
      *
-     * @param content the text to escape, may be {@code null}
+     * @param content the text to escape, may be {@code null}. The content is converted to {@code String} using
+     * {@link String#valueOf(Object)} before escaping.
      * @return a new escaped {@code String}, {@code null} if {@code null} input
+     * @deprecated since 12.8RC1, use {@link #escapeElementContent(String)} instead.
      */
+    @Deprecated
     public static String escapeElementContent(Object content)
     {
         if (content == null) {
             return null;
         }
-        String str = String.valueOf(content);
-        StringBuilder result = new StringBuilder((int) (str.length() * 1.1));
-        int length = str.length();
+        return escapeElementContent(String.valueOf(content));
+    }
+
+    /**
+     * Escapes the XML special characters in a <code>String</code> using numerical XML entities, so that the resulting
+     * string can safely be used as an XML text node. Specifically, escapes &lt;, &gt;, and &amp;.
+     *
+     * @param content the text to escape, may be {@code null}
+     * @return a new escaped {@code String}, {@code null} if {@code null} input
+     * @since 12.8RC1
+     */
+    @Unstable
+    public static String escapeElementContent(String content)
```

Comment:
It's not clear at all from the JavaDoc **why** we need both ``escapeElementText`` and ``escapeElementContent``. The JavaDoc explains what each method does and I understand that the only difference is that ``escapeElementText`` doesn't escape ``>`` (``&gt;``) but when would you use one versus the other? When is it OK to not escape ``>`` and when is it mandatory to escape it?

Reply:
Actually escaping `>` is never really needed so I don't understand why we need escapeElementContent at all.

Reply:
Would it be ok to deprecate `escapeElementContent` and suggest to use `escapeElementText` instead?

Reply:
Sounds good.

Reply:
Hum, in this case we are in a odd situation when `public static String escapeElementContent(String content)` will be both `@Unstable` and `@Deprecated`. Does deprecated takes precedence on unstable?

Reply:
I don't understand what you mean, `escapeElementContent(String content)` should not exist at all.

Reply:
`escapeElementContent` deprecated and `escapeElementText` recommended instead